HerPublicWebsite.BusinessLogic/Services/RegularJobs/PolicyTeamUpdateService.cs

```diff
@@ -0,0 +1,61 @@
+using HerPublicWebsite.BusinessLogic.ExternalServices.EmailSending;
+using HerPublicWebsite.BusinessLogic.Services.WorkingDayHelper;
+
+namespace HerPublicWebsite.BusinessLogic.Services.PolicyTeamUpdate;
+
+public interface IPolicyTeamUpdate
+{
+    public Task SendPolicyTeamUpdate();
+}
+
+public class PolicyTeamUpdateService : IPolicyTeamUpdate
+{
+    private readonly IDataAccessProvider dataProvider;
+    private readonly CsvFileCreator.CsvFileCreator csvFileCreator;
+    private readonly IWorkingDayHelperService workingDayHelperService;
+    private readonly IEmailSender emailSender;
+    
+    public PolicyTeamUpdateService(
+        IDataAccessProvider dataProvider,
+        CsvFileCreator.CsvFileCreator csvFileCreator, 
+        IWorkingDayHelperService workingDayHelperService,
+        IEmailSender emailSender
+    )
+    {
+        this.dataProvider = dataProvider;
+        this.csvFileCreator = csvFileCreator;
+        this.workingDayHelperService = workingDayHelperService;
+        this.emailSender = emailSender;
+    }
+
+    public async Task SendPolicyTeamUpdate(){
+        var recentReferralRequestOverviewTable = await BuildRecentReferralRequestOverviewTable();
+        var recentReferralRequestFollowUpTable = await BuildRecentReferralRequestFollowUpTable();
+        var historicReferralRequestFollowUpTable = await BuildHistoricReferralRequestFollowUpTable();
+        emailSender.SendComplianceEmail(
+            recentReferralRequestOverviewTable,
+            recentReferralRequestFollowUpTable,
+            historicReferralRequestFollowUpTable
+        );
+
+    }
+
+    private async Task<MemoryStream> BuildRecentReferralRequestOverviewTable (){
+        DateTime endDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now, -10); 
+        DateTime startDate = endDate.AddDays(-(int)DateTime.Today.DayOfWeek + (int)DayOfWeek.Monday);
+        var newReferrals = await dataProvider.GetReferralRequestsBetweenDates(startDate, endDate);
+        return csvFileCreator.CreateReferralRequestOverviewFileData(newReferrals);
+    }
+
+    private async Task<MemoryStream> BuildRecentReferralRequestFollowUpTable (){
+        DateTime endDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now, -10); 
+        DateTime startDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Now.AddDays(-7), -10); 
```

Comment:
This assumes that the update never fails which seems a bit risky. If it does then it will miss out data for a week. I don't know if we want to guard against that somehow?

Could add a column to the referrals table saying it's been included in a policy team update which we switch whenever a row is included in an update I suppose?

Reply:
Good point. I think like you say, as we do have our historic data statistics we should be OK with this. If this fails and the Referral Requests are required, re-running the policy email generator with the missed dates would probably be a better solution anyway.